cdap-docs/examples-manual/source/examples/data-cleansing.rst

```diff
@@ -0,0 +1,178 @@
+---------------------------------
+
+``DataCleansingMapReduce`` is a simple MapReduce that reads from the ``inputRecords`` PartitionedFileSet and writes to
+the ``outputRecords`` PartitionedFileSet. The ``beforeSubmit()`` method prepares the MapReduce program for this:
+
+- It uses the ``BatchPartitionConsumer`` to specify the partitions to process as input, in order to only process new
+  partitions since its last run.
+- It specifies the output partition to write to based upon the runtime arguments.
+
+
+Building and Starting
+=====================
+
+.. include:: building-and-starting.txt
+
+
+Running CDAP Applications
+=========================
+
+.. |example| replace:: DataCleansing
+
+.. include:: /../../developers-manual/source/getting-started/building-apps.rst
+   :start-line: 11
+
+
+Running the Example
+===================
+
+Starting the Service
+--------------------
+
+Once the application is deployed:
+
+- Go to the *DataCleansing* `application overview page
+  <http://localhost:9999/ns/default/apps/DataCleansing/overview/status>`__,
+  click ``DataCleansingService`` to get to the service detail page, then click the *Start* button; or
+- From the Standalone CDAP SDK directory, use the Command Line Interface::
+
+    $ cdap-cli.sh start service DataCleansing.DataCleansingService
+
+    Successfully started service 'DataCleansingService' of application 'DataCleansing' with stored runtime arguments '{}'
+
+Ingesting Records
+-----------------
+
+Begin by uploading a file containing some newline-separated JSON records into the ``inputRecords`` dataset::
+
+  $ cdap-cli.sh call service DataCleansing.DataCleansingService PUT inputRecords body:file examples/DataCleansing/resources/person.json
+
+Starting the MapReduce
+----------------------
+
+The MapReduce must be started with a runtime argument 'output.partition.key' in order to specify the output partition of
+the 'outputRecords' dataset to write to. In this example, we'll simply use 1 as the value.
+- Go to the *DataCleansing* `application overview page
+  <http://localhost:9999/ns/default/apps/DataCleansing/overview/status>`__,
+  click ``DataCleansingMapReduce`` to get to the MapReduce detail page, set the runtime
+  arguments 'output.partition.key' = 1, then click the *Start* button; or
+- Use the Command Line Interface::
+
+    $ cdap-cli.sh start mapreduce DataCleansing.DataCleansingMapReduce output.partition.key=1
+
+    Successfully started mapreduce 'DataCleansingMapReduce' of application 'DataCleansing' with provided runtime arguments 'output.partition.key=1'
+
+Optionally, to specify a custom schema to match records against, the JSON of the schema can be
+specified as an additional runtime argument to the MapReduce with the key 'schema.key'.
+Otherwise, the default schema that is matched against the records:
+
+.. literalinclude:: /../../../cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduce.java
```

Comment:
This literal include should be added to the build.sh with an updated hash, as per above instructions.